Sources/UIKit/TextAlignmentConstraint.swift

```diff
@@ -0,0 +1,162 @@
+    }
+
+    public override func awakeFromNib() {
+        super.awakeFromNib()
+        setupObservers()
+    }
+
+    private func setupObservers() {
+        for keyPath in fontKeyPaths {
+            addObserver(self, forKeyPath: keyPath, options: [], context: &TextAlignmentConstraintKVOContext)
+        }
+    }
+
+    private func tearDownObservers() {
+        for keyPath in fontKeyPaths {
+            removeObserver(self, forKeyPath: keyPath, context: &TextAlignmentConstraintKVOContext)
+        }
+    }
+
+    private var fontKeyPaths: [String] {
+        let firstItemFontKeyPath = [
+            #selector(getter: NSLayoutConstraint.firstItem),
+            FontSelector
+            ].map({ $0.description })
+            .joined(separator: ".")
+
+        let secondItemFontKeyPath = [
+            #selector(getter: NSLayoutConstraint.secondItem),
+            FontSelector
+            ].map({ $0.description })
+            .joined(separator: ".")
+
+        return [firstItemFontKeyPath, secondItemFontKeyPath]
+    }
+
+    public override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        guard context == &TextAlignmentConstraintKVOContext else {
+            super.observeValue(forKeyPath: keyPath, of: object, change: change, context: context)
+            return
+        }
+
+        updateConstant()
+    }
+
+    private func updateConstant() {
+        let distanceFromTop1 = distanceFromTop(of: item1, with: firstItemAttribute)
+        let distanceFromTop2 = distanceFromTop(of: item2, with: secondItemAttribute)
+        let difference = distanceFromTop2 - distanceFromTop1
+        constant = difference
+    }
+
+    private func distanceFromTop(of item: AnyObject, with attribute: TextAlignmentConstraintAttribute) -> CGFloat {
+        guard let font = font(from: item) else {
+            return 0
+        }
+
+        let topToBaseline = font.ascender
+        let distanceFromTop: CGFloat
+
+        switch attribute {
+        case .capHeight:
+            distanceFromTop = topToBaseline - font.capHeight
+        case .xHeight:
+            distanceFromTop = topToBaseline - font.xHeight
+        case .top:
+            distanceFromTop = 0
+        case .firstBaseline, .lastBaseline, .bottom:
+            fatalError("TODO: write an error")
+        case .unspecified:
+            fatalError("TODO: write an error")
+        }
+
+        return distanceFromTop
+    }
+
+    private func font(from item: AnyObject) -> BONFont? {
+        var font: BONFont? = nil
+
+        if item.responds(to: FontSelector) {
+            font = item.perform(FontSelector).takeUnretainedValue() as? BONFont
+        }
+
+        return font
+    }
+}
+
+// The class part of this selector is ignored; it is there simply to satisfy Xcode's new selector syntax.
+#if os(OSX)
+    private let FontSelector = #selector(getter: NSTextField.font)
```

Comment:
You are correct. Boo. Good comment, then. I'd nest it as a `static let` inside the constraint subclass, and use a lowercase initial letter.